firecracker/rootfs/generate_firecracker_rootfs.sh

```diff
@@ -0,0 +1,24 @@
+#!/usr/bin/env bash
+set -euo pipefail
+
+THIS_DIR=$(dirname "${BASH_SOURCE[0]}")
+readonly THIS_DIR
+
+REPOSITORY_ROOT=$(git rev-parse --show-toplevel)
+readonly REPOSITORY_ROOT
+
+(
+    cd "${THIS_DIR}"
+    docker buildx bake -f bake.hcl rootfs-build
+)
+
+########################################
+# Generate the rootfs.
+# We use Fuse to mount inside Docker.
+########################################
+docker run \
+    --rm \
+    --device /dev/fuse \
+    --cap-add SYS_ADMIN \
+    --volume "${REPOSITORY_ROOT}/dist:/dist" \
+    rootfs-build:dev
```

Comment:
I'm curious why you need to build an image and then run it to make the rootfs tarball... could this all be done in the image build itself and then copied out using the `output =["type=local,dest=."]` trick in the bake HCL file?

Also curious why you need `fuse`.

Reply:
# Why fuse
We need `fuse` because mounting loop devices isn't really supported on crostini:
https://grapl-internal.slack.com/archives/C0174A8QV2S/p1645807029980149?thread_ts=1645720167.748819&cid=C0174A8QV2S

I believe "an image file" qualifies as a loop device.

# Why not at `bake` time?
Since we need fuse, we need `--device /dev/fuse` to use it from within Docker.
Since we need `--device /dev/fuse` - and `--cap-add SYS_ADMIN` - I presumably can't do that with `buildx bake`. I can't prove that they're not supp

Reply:
It looks like you're wanting to create a directory structure and then package that up as a file; I'm just wondering if you can do that by literally creating the directory structure you want and then feeding that into a process that creates the desired file format. If so, that seems like it might be simpler for this usecase.

If not, that's fine!

Reply:
@wimax-grapl I think it would be useful to take the Why fuse section here and put it in either a comment or in a doc.

Reply:
I haven't been able to find such a process-that-creates-ext4; I was mostly going off of [this firecracker guide](https://github.com/firecracker-microvm/firecracker/blob/main/docs/rootfs-and-kernel-setup.md). 

I believe the "format," so to speak, is a "KVM image"

Reply:
https://unix.stackexchange.com/questions/503211/how-can-an-image-file-be-created-for-a-directory

Looks like what Max is doing is probably the best bet